sklearn/random_projection.py

```diff
@@ -431,6 +445,26 @@ def _n_features_out(self):
         """
         return self.n_components
 
+    @available_if(lambda self: self.fit_inverse_transform)
+    def inverse_transform(self, X):
+        """Project data back to its original space.
+
+        Returns an array X_original whose transform would be X.
+
```

Comment:
I added "Note that even if `X` is sparse, `X_original` is dense: this may use a lot of RAM."

Reply:
Maybe we would worth adding a note that users interested in recovering a sparse `X_original` that matches the sparsity level of `X_train` could use `SparseCoder(transform_algorithm="omp")` with `transform_n_nonzero_coefs` set to `X_train.nnz / X_train.shape[0]`.

https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.SparseCoder.html

However I have no idea how tractable this approach is on realistic dataset sizes.

Reply:
Edit: I realize that `SparseCoder` will always return dense arrays so memory usage remain challenging. It would be possible to convert to sparse matrices in chunks and then concatenate the chunks at the end to avoid blowing up the memory but this would be much more involved.

Maybe it's too complicated and we should not mention `SparseCoder` at all in retrospect.

Reply:
Still maybe we could store a private:

```python
    self._sparse_X_train = sp.issparse(X)
```

at fit time and then issue a warning in `inverse_transform` to warn about potentially large memory usage in this case prior to calling the projection.